### PR TITLE
Material system documentation + minor refactoring wrt Radium Guidelines

### PR DESCRIPTION
**Pull request by MathiasPaulin**
* **Please check if the PR fulfills these requirements**
- [X] Is the Pull-Request done against the right branch:
  - `master-v1`: for changes related to the Radium Stable Release V1.
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Documentation of the Material subsystem of the Engine and how it could be extended by plugins or applications.

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No API changes. just moving some classes between files. users must only recompile.

* **Other information**:

### CODE REVIEW

File: src/Engine/Renderer/Material/BlinnPhongMaterial.hpp [hunk 1]
```diff
@@ -8,12 +8,16 @@
 
 #include <Engine/Renderer/Material/Material.hpp>
 
-// just needed for struct TextureData
+// just need for struct TextureData
```
Comment (dlyr) [inline src/Engine/Renderer/Material/BlinnPhongMaterial.hpp]: ? are you sure ;)
https://translate.google.com/#en/fr/just%20needed%20for%20struct%20TextureData

Comment (MathiasPaulin) [reply src/Engine/Renderer/Material/BlinnPhongMaterial.hpp]: Yes, I'm sure ... we only need to have access to the definition of TextureData

Comment (dlyr) [reply src/Engine/Renderer/Material/BlinnPhongMaterial.hpp]: But are you sure the tense is "need" and not "needed" ?
BTW I don't understand the comment, do you mean an include here has been removed or that

Comment (MathiasPaulin) [reply src/Engine/Renderer/Material/BlinnPhongMaterial.hpp]: I mean that this include is only necessary to access the definition of the struct TextureData. For this little thing, we bring a lot of others not needed ... BTW, we can remove this comment.

**Comment by dlyr** [comment]
And remove the forward declaration of texturedata ?

**Pull request merged**